src/features/favoritesSlice.js

```diff
@@ -8,10 +8,51 @@ export const favoritesSlice = createSlice({
     name: "favorites",
     initialState,
     reducers: {
-        updateFavorites: (state, action) => {
-            state.data = action.payload;
+        // updateFavorites: (state, action) => {
+        //     localStorage.setItem("favoriteCoinIds", action.payload);
+        //     state.data = action.payload;
+        // },
+        addNewFavorite: (state, action) => {
+            // expected payload: new coin id
+            // make a copy of state, spread in the new coin
+            const copiedState = [...state.data, action.payload];
+            // add copy with new coin to LS
+            // update redux state with copy with new coin
+            localStorage.setItem("favoriteCoinIds", action.payload);
+            copiedState.data = action.payload;
+            localStorage.setItem("favoriteCoinIds", copiedState);
+
+            // have to check if coin already exists in LS "favoriteCoinIds", prior to adding it again.
```

Comment:
This block of comments should be transferred to trello as another card or something. It doesnt belong here. This logic should also be taken care of at the input in main content, not in the reducer. We don't want this sort of logic making it this far into the process. We have the ability of knowing at the input value collection point, we should handle it as early in the process as possible